### PR TITLE
Add java version and configure to VersionInfo

### PR DESCRIPTION
**Pull request by M-Davies**
* Adds the output of java --version and bash configure to VersionInfo
* Adds version tests to account for the changes
* Documentation updates

Closes: #11 
Closes: #367 
Signed-off-by: Morgan Davies <morgan.davies@ibm.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -323,16 +323,19 @@ Alongside the built assets a metadata file will be created with info about the b
...
             "security": 202,
             "build": 8,
             "version": "8u202-b08",
             "semver": "8.0.202+8.2",
+            "full_version_output": "<output of java --version>",
+            "configure_args": "<output of bash configure>"
         },
-        "binary_type": "jdk"
+        "binary_type": "jdk",
+        "sha256": "c1b8fb7298d66a5bca9b830e8d612a85bbc52c81b9a88cef4dd71f2f37b289f1"
```
Comment (karianna) [inline README.md:338]: perhaps <output of the shasum>

File: pipelines/build/common/openjdk_build_pipeline.groovy [hunk 2]
```diff
@@ -400,27 +400,29 @@ class Build {
...
+                "minor": 0,
+                "security": 202,
+                "build": 8,
+                "version": "8u202-b08",
+                "semver": "8.0.202+8.2",
+                "full_version_output": "<output of java --version>",
+                "configure_args": "<output of bash configure>"
+            },
+            "binary_type": "jdk",
+            "sha256": "c1b8fb7298d66a5bca9b830e8d612a85bbc52c81b9a88cef4dd71f2f37b289f1"
```
Comment (karianna) [inline pipelines/build/common/openjdk_build_pipeline.groovy:423]: as abobe <output of shasum>

File: pipelines/src/test/groovy/VersionInfoTest.groovy [hunk 3]
```diff
@@ -45,4 +45,38 @@ class VersionInfoTest {
         Assertions.assertEquals("11.0.2+10.2", parsed.semver)
     }
 
+    def versionOut = """openjdk version "11.0.2" 2018-10-16
+OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.2+7)
+OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.2+7, mixed mode)"""
+
+    @Test
+    void addsVersionOut() {
```
Comment (karianna) [inline pipelines/src/test/groovy/VersionInfoTest.groovy:53]: addsFullVersionOutput

File: pipelines/src/test/groovy/VersionInfoTest.groovy [hunk 4]
```diff
@@ -45,4 +45,38 @@ class VersionInfoTest {
...
+        Assertions.assertEquals(versionOut, parsed.full_version_output)
+        Assertions.assertNull(parsed.opt)
+        Assertions.assertEquals("11.0.2+7", parsed.version)
+        Assertions.assertNull(parsed.pre)
+        Assertions.assertNull(parsed.adopt_build_number)
+        Assertions.assertEquals("11.0.2+7", parsed.semver)
+    }
+
+    @Test
+    void addsNullFullVersionOutputWhenThereIsNotOne() {
```
Comment (karianna) [inline pipelines/src/test/groovy/VersionInfoTest.groovy:68]: AddsNullWhenNoFullVersionOutput

File: README.md (hunk 1 above)
Comment (M-Davies) [reply README.md:338]: Perhaps?

File: pipelines/src/test/groovy/VersionInfoTest.groovy (hunk 3 above)
Comment (M-Davies) [reply pipelines/src/test/groovy/VersionInfoTest.groovy]: Done

File: pipelines/src/test/groovy/VersionInfoTest.groovy (hunk 4 above)
Comment (M-Davies) [reply pipelines/src/test/groovy/VersionInfoTest.groovy]: Done

File: README.md (hunk 1 above)
Comment (karianna) [reply README.md:338]: Ugh, github was not playing nice today.  Perhaps make that a &lt;sha sam&gt; as opposed to putting in a real value

Comment (M-Davies) [reply README.md]: Done

**Comment by karianna** [comment]
LGTM if it's ready to be taken out of draft mode

**Comment by M-Davies** [comment]
Still trying to get it working at https://ci.adoptopenjdk.net/view/work%20in%20progress/job/jdk15-mac-x64-hotspot/2/console

**Comment by M-Davies** [comment]
https://ci.adoptopenjdk.net/view/work%20in%20progress/job/jdk15-mac-x64-hotspot/4/ is now working (see the [JSON file](https://ci.adoptopenjdk.net/view/work%20in%20progress/job/jdk15-mac-x64-hotspot/4/artifact/workspace/target/OpenJDK15-jdk_x64_mac_hotspot_2020-06-22-09-05.pkg.json) for the `full_version_output` and `configure_arguments`). `configure_arguments` also get passed to a [standalone file](https://ci.adoptopenjdk.net/view/work%20in%20progress/job/jdk15-mac-x64-hotspot/4/artifact/workspace/target/configure.txt) incase it's ever needed.

**Comment by karianna** [comment]
I think in your example you call it version_data and in the sample above version...

**Comment by M-Davies** [comment]
Ah that'll be the outdate README. I'll fix it

**Comment by M-Davies** [comment]
[JDK8](https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk8-pipeline/1132/) and [JDK11](https://ci.adoptopenjdk.net/job/build-scripts/job/openjdk11-pipeline/1027/) test pipelines kicked off

**Pull request merged**